glusterd2/commands/peers/addpeer.go

```diff
@@ -25,6 +26,14 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	for key := range req.Metadata {
+		if strings.HasPrefix(key, "_") {
+			logger.WithField("metadata-key", key).Error("Key names starting with '_' are restricted in metadata field")
```

Comment:
This error can be added to generic errors list. Used this in multiple places and also Volume metadata will also requires this error msg.